### PR TITLE
add support for %%env_%% AD template tag

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

As discussed in https://github.com/DataDog/datadog-agent/issues/1250, add a new tag in AD templates, to inject environment variables during template resolution.

This allows to inject kubernetes downwards API information, or credentials, in templates.

For example, if I want to monitor a pod running on host networking, and already injected my node's IP in the `$KUBERNETES_KUBELET_HOST` envvar, I can use the following template:

```yaml
ad_identifiers:
  - redis

init_config:

instances:
  - host: "%%env_KUBERNETES_KUBELET_HOST%%"
    port: "1234"
```

### CODE REVIEW

File: pkg/collector/autodiscovery/common_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package autodiscovery
+
+import "github.com/DataDog/datadog-agent/pkg/collector/listeners"
+
+type dummyService struct {
```
Comment (mfpierre) [inline pkg/collector/autodiscovery/common_test.go]: nice addition for testing 👍

**Review by mfpierre — CHANGES_REQUESTED** [review]
Nice stuff 👏 

I think we're just missing a reno note and a readme update here https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/listeners/README.md

**Review by mfpierre — APPROVED** [review]
🍰

**Comment by xvello** [comment]
@l0k0ms any idea where to best document this new feature?

**Comment by l0k0ms** [comment]
@xvello either here: https://github.com/DataDog/documentation/blob/master/content/agent/autodiscovery.md#setting-up-check-templates

or https://github.com/DataDog/documentation/blob/master/content/agent/autodiscovery.md#example-apache-check

or maybe in all AD templates?

**Comment by xvello** [comment]
Yup, so we don't currently have a table of supported template vars.
Do you think I could add a section to
https://github.com/DataDog/documentation/blob/master/content/agent/autodiscovery.md#reference
?

**Comment by l0k0ms** [comment]
yup it's a really good idea! I'm here if you need help for it

**Comment by pussinboots** [comment]
Thanks for that feature. That make it possible to get ride of some awful bash script replace code. 
👍

**Pull request merged**